packages/libircclient/patch-add-cloudabi

```diff
@@ -0,0 +1,451 @@
+--- configure
++++ configure
+@@ -623,6 +623,7 @@ ac_includes_default="\
+ #endif"
+ 
+ ac_subst_vars='LTLIBOBJS
++BUILD_EXAMPLES
+ PREFIX
+ TARGET
+ LIBOBJS
+@@ -687,6 +688,7 @@ enable_shared
+ enable_threads
+ enable_ipv6
+ enable_openssl
++enable_examples
+ '
+       ac_precious_vars='build_alias
+ host_alias
+@@ -1314,6 +1316,7 @@ Optional Features:
+   --enable-threads        compile with multithread support (yes)
+   --enable-ipv6           compile with IPv6 support (no)
+   --enable-openssl        compile with OpenSSL support (no)
++  --enable-examples       compile with examples (yes)
+ 
+ Some influential environment variables:
+   CXX         C++ compiler command
+@@ -2247,6 +2250,14 @@ else
+ fi
+ 
+ 
++# Check whether --enable-examples was given.
++if test "${enable_examples+set}" = set; then :
++  enableval=$enable_examples; with_examples=$enableval
++else
++  with_examples=yes
++fi
++
++
+ # Checks for programs.
+ ac_ext=cpp
+ ac_cpp='$CXXCPP $CPPFLAGS'
+@@ -4535,6 +4546,10 @@ if test "$use_threads" = "yes"; then
+ 	CFLAGS="$CFLAGS -DENABLE_THREADS -D_REENTRANT"
+ fi
+ 
++if test "$with_examples" = "yes"; then
++	BUILD_EXAMPLES=1
++fi
++
+ 
+ # Checking for mingw
+ { $as_echo "$as_me:${as_lineno-$LINENO}: checking For MinGW32" >&5
+@@ -4551,6 +4566,19 @@ $as_echo "yes" >&6; }
+ $as_echo "no" >&6; } ;;
+ esac
+ 
++# Checking for cloudabi
++{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for CloudABI" >&5
++$as_echo_n "checking for CloudABI... " >&6; }
++case "$host" in
++	*-cloudabi*)
++		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
++$as_echo "yes" >&6; }
++		CFLAGS="$CFLAGS -DCLOUDABI"
++		;;
++	*) { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
++$as_echo "no" >&6; } ;;
++esac
++
+ if test "$use_openssl" = "yes"; then
+ 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for CRYPTO_new_ex_data in -lcrypto" >&5
+ $as_echo_n "checking for CRYPTO_new_ex_data in -lcrypto... " >&6; }
+@@ -4662,6 +4690,7 @@ fi
+ 
+ 
+ 
++
+ ac_config_files="$ac_config_files examples/Makefile src/Makefile"
+ 
+ cat >confcache <<\_ACEOF
+--- configure.in
++++ configure.in
+@@ -37,6 +37,12 @@ AC_ARG_ENABLE([openssl],
+             [use_openssl=$enableval],
+             [use_openssl=no])
+ 
++AC_ARG_ENABLE([examples],
++            [AS_HELP_STRING([--enable-examples],
++               [compile with examples (yes)])],
++            [with_examples=$enableval],
++            [with_examples=yes])
++
+ # Checks for programs.
+ AC_PROG_CXX
+ AC_PROG_CC
+@@ -86,6 +92,10 @@ if test "$use_threads" = "yes"; then
+ 	CFLAGS="$CFLAGS -DENABLE_THREADS -D_REENTRANT"
+ fi
+ 
++if test "$with_examples" = "yes"; then
++	BUILD_EXAMPLES=1
++fi
++
+ 
+ # Checking for mingw
+ AC_MSG_CHECKING([For MinGW32])
+@@ -99,6 +109,16 @@ case "$host" in
+ 	*) AC_MSG_RESULT([no]) ;;
+ esac
+ 
++# Checking for cloudabi
++AC_MSG_CHECKING([for CloudABI])
++case "$host" in
++	*-cloudabi*)
++		AC_MSG_RESULT([yes])
++		CFLAGS="$CFLAGS -DCLOUDABI"
```

Comment:
I just thought it'd be nicer if the build system had some control over it. But then again, it would have been better to add HAS_CONNECT and HAS_SOCKET in that case. I guess it's better to use `__CloudABI__` now that I think about it.